src/k8s-engine/wsl.ts

```diff
@@ -512,6 +527,55 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     }
   }
 
+  /**
+   * Write out /etc/hosts in the main distribution, copying the bulk of the
+   * contents from the data distribution.
+   */
+  protected async writeHostsFile() {
+    await this.progressTracker.action('Updating /etc/hosts', 50, async() => {
+      const contents = await fs.promises.readFile(`\\\\wsl$\\${ DATA_INSTANCE_NAME }\\etc\\hosts`);
+      const hosts = ['host.rancher-desktop.internal', 'host.docker.internal'];
+      const extra = [
+        '# BEGIN Rancher Desktop configuration.',
+        `${ this.hostIPAddress } ${ hosts.join(' ') }`,
+        '# END Rancher Desktop configuration.',
+      ].map(l => `${ l }\n`).join('');
+
+      await fs.promises.writeFile(`\\\\wsl$\\${ INSTANCE_NAME }\\etc\\hosts`,
+        Buffer.concat([contents, Buffer.from(extra, 'utf-8')]));
+    });
+  }
+
+  /**
+   * Write configuration for dnsmasq / and /etc/resolv.conf; required before [runInit].
+   */
+  protected async writeResolvConf() {
+    await this.progressTracker.action('Updating DNS configuration', 50,
+      // Tell dnsmasq to use the resolv.conf from the data distro as the
+      // upstream configuration.
+      Promise.all([
+        (async() => {
+          try {
```

Comment:
The read can fail if the data distro isn't ready.  The try block is here because it is useful to get a better error message here (especially because it involves a different distro).  The others can fail with the default message which we are likely to interpret as relating to the main distro.

(That is to say: this was added during debugging this PR, because it was confusing when an error occurred.)